src/core/core/network.cpp

```diff
@@ -340,6 +340,9 @@ namespace dsn
     {
         if (is_client())
         {
+            /* TODO(qinzuoyan): now that we will always call on_recv_reply() to notice the failure,
```

Comment:
I means that the users can get connection failure immediately and do retry if they want, then reconnect is not so important. And, reconnect using a closed socket may cause some problems under asio, which I have meet before.